browser/ai_chat/tools/target_util.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/tools/target_util.h"
+
+#include "base/notreached.h"
+#include "brave/components/ai_chat/core/browser/tools/tool_input_properties.h"
+#include "chrome/common/actor/actor_constants.h"
+
+namespace ai_chat::target_util {
+
+base::Value::Dict TargetProperty(const std::string& description) {
+  base::Value::Dict target_property;
+  target_property.Set("description", description);
+  base::Value::List* any_of = target_property.EnsureList("anyOf");
+
+  any_of->Append(ObjectProperty(
+      "DOM element identifiers of target (preferred)",
+      {{"document_identifier",
+        StringProperty("Document identifier for the target frame")},
```

Comment:
Sorry not 100% clear to me what this can be based on reading this. Are only `id`s valid or can this be any CSS selector?